### DISCUSSION TITLE
可以考虑将poc加密后再硬编码到template.go中

### DISCUSSION THREAD

**Discussion by lovelyjuice** [chainreactors/gogo#95] [Ideas]
生成模板的时候使用对称密钥加密。
使用的时候需要加`--tpl-key xxxxxx`的参数作为解密密钥，或者从文件、url中获取密钥，之后gogo在内存中解密模板内容

有以下两个好处：
1. 减少静态免杀压力
2. 在非运行状态，即便被其它人拿到二进制文件（比如杀软上传到云、被传到VT，或是被其它红队拿到），也不会导致内部poc泄露

**Comment by M09Ic** [comment]
是个好主意, 其实gogo曾经提供过类似的功能, 现在还保留了一个不在文档中的flag	`--key`. 下个版本会考虑重新启用这个功能, 并应用到全局
